### ISSUE TITLE
empty set raises problems

### ISSUE THREAD

**Issue by benmaier** [gstonge/SamplableSet#31]
Hey Guillaume!
Constructing an empty set seems to cause problems now. see here:

```python
In [1]: from SamplableSet import SamplableSet

In [2]: min_weight = 1; max_weight=1; neighbors = {};

In [3]: SamplableSet(min_weight,max_weight,neighbors)
Out[3]: ---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/usr/local/lib/python3.8/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

/usr/local/lib/python3.8/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    392                         if cls is not object \
    393                                 and callable(cls.__dict__.get('__repr__')):
--> 394                             return _repr_pprint(obj, self, cycle)
    395
    396             return _default_pprint(obj, self, cycle)

/usr/local/lib/python3.8/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    698     """A pprint that just redirects to the normal repr function."""
    699     # Find newlines and replace them with p.break_()
--> 700     output = repr(obj)
    701     lines = output.splitlines()
    702     with p.group():

/usr/local/lib/python3.8/site-packages/SamplableSet/_wrapper.py in __repr__(self)
    136
    137     def __repr__(self):
--> 138         return str(self)
    139
    140     def __len__(self):

/usr/local/lib/python3.8/site-packages/SamplableSet/_wrapper.py in __str__(self)
    133
    134     def __str__(self):
--> 135         return f'SamplableSet of {len(self)} element' + ('s' if len(self) > 1 else '')
    136
    137     def __repr__(self):

/usr/local/lib/python3.8/site-packages/SamplableSet/_wrapper.py in __len__(self)
    139
    140     def __len__(self):
--> 141         return self.size()
    142
    143     def copy(self):

AttributeError: 'SamplableSet' object has no attribute 'size'
```

I think this worked before because cpp_type=int was implicit but now it tries to infer the type? Is this behavior intentional? 

btw, it works fine when I  call

```python
SamplableSet(min_weight,max_weight,{},cpp_type="int")
```
Cheers!

**Comment by gstonge** [comment]
Oh yeah, good catch! The problem is actually in the `__str__` method when the elements' type has not been inferred yet.
For instance, this would work in your example:
`s = SamplableSet(min_weight,max_weight,neighbors)`
I'll fix it quickly.

**Comment by gstonge** [comment]
Fixed by #32
